lib/mqttclient.ts

```diff
@@ -315,7 +315,8 @@ export default class MQTTClient {
                         },
                         name: obj.name,
                         type: obj.type,
-                        alias: alias
+                        alias: alias,
+                        transient: !(obj.properties?.recordToDB?.value)
```

Comment:
* I think `recordToDB` is our local invention; IIRC it was a hack for the Ignition historian backend because Ignition doesn't expose the Sparkplug `is_transient` metric field to its internal scripting. We should be using `is_transient` directly going forward, and probably drop `recordToDB` altogether.
* This defaults to not recording metrics which don't specify `recordToDB` at all (opt-in). Switching to `is_transient` will also switch to an opt-out model, where publishers have to explicitly request not to be recorded. The F+ services which publish over MQTT (Directory, ConfigDB) currently don't set either `recordToDB` or `is_transient`; we should think about whether their output should be recorded in a historian, or not.